packages/ui/components/form/select/SelectDownshift.tsx

```diff
@@ -0,0 +1,248 @@
+          }
+          break;
+        default:
+          break;
+      }
+    },
+  });
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    getInputProps,
+    getItemProps,
+    highlightedIndex,
+  } = useCombobox({
+    items,
+    stateReducer(state, actionAndChanges) {
+      const { changes, type } = actionAndChanges;
+
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+        case useCombobox.stateChangeTypes.InputBlur:
+          return {
+            ...changes,
+            ...(changes.selectedItem && { isOpen: true, highlightedIndex: changes.highlightedIndex }),
+          };
+        default:
+          return changes;
+      }
+    },
+    onStateChange({ type, selectedItem: newSelectedItem, inputValue: newInputValue }) {
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+          if (newSelectedItem) {
+            if (selectedItems.includes(newSelectedItem)) {
+              removeSelectedItem(newSelectedItem);
+              props.onSelectedItemsChange && props.onSelectedItemsChange(selectedItems);
+              return;
+            }
+            const newSelectedItems = [...selectedItems, newSelectedItem];
+            setSelectedItems(newSelectedItems);
+            props.onSelectedItemsChange && props.onSelectedItemsChange(newSelectedItems);
+          }
+          break;
+
+        case useCombobox.stateChangeTypes.InputChange:
+          setInputValue(newInputValue || "");
+
+        default:
+          break;
+      }
+    },
+  });
+
+  return (
+    <>
+      <div className="flex flex-col">
+        <Label {...getLabelProps()}>Pick some books:</Label>
+        <div
+          aria-label="menu-toggle"
+          {...getToggleButtonProps(getDropdownProps({ preventKeyAction: isOpen }))}
+          className="flex h-9 space-x-1 rounded-md border border-gray-300 p-1 focus-within:border-gray-900 hover:cursor-pointer hover:border-gray-400 focus-visible:border-gray-900">
+          {!props.renderNode
+            ? selectedItems.map(function renderSelectedItem(selectedItemForRender, index) {
+                return (
+                  <div
+                    className="hocus:text-gray-900 hocus:bg-gray-300 focus:border-1 flex items-center rounded-md border-transparent bg-gray-200 px-2 py-[6px] text-sm text-gray-700"
+                    key={`selected-item-${index}`}
+                    {...getSelectedItemProps({
+                      selectedItem: selectedItemForRender,
+                      index,
+                    })}>
+                    <span>
+                      <>{selectedItemForRender[props.labelKey]}</>
+                    </span>
+                    <span
+                      className="ml-2 cursor-pointer"
```

Comment:
I guess this should be a button with an a11y readable text?

Reply:
I'm pretty sure the 
```
  {...getSelectedItemProps({
        selectedItem: selectedItemForRender,
        index,
      })}>
```
Adds an aria label to this div so likely not needed?

Reply:
Ooooh, that could very well be indeed. Isn't it a bit weird then that the onclick is on the span and not the div? I guess the div should be clickable then (and become the button)?

Reply:
Theres a weird reason why its a div and not a button if i remember what the docs were saying... Something to do with screen reader priority

And it needs to be on the whole div as this handles keyboard controls on the whole item which is as per the a11y spec 

e.g. pressing backspace/del will remove it when keyboard focused